code/parachain/frame/composable-traits/src/xcm/assets.rs

```diff
@@ -89,3 +89,38 @@ pub struct ForeignMetadata<AssetNativeLocation> {
 	pub decimals: Option<Exponent>,
 	pub location: AssetNativeLocation,
 }
+
+pub trait PermissionlessAsset {
+	type AccountId;
+	type AssetId;
+	type AssetNativeLocation;
+	type Balance;
+
+	fn register_asset(
+		account_id: Self::AccountId,
+		asset_id: Self::AssetId,
+		location: Self::AssetNativeLocation,
```

Comment:
i guess this is not needed here. `location` can be registered only by other chain governance, not permissionless.

Reply:
and also it allows routing some other chain assets under admin and mint burn, so can still.

Reply:
If location is used it won't be allowed to be used again. I think rn the idea was that user manually creates asset that they want to transfer. For that they will need location.

Reply:
I think it should be ok to have location supplied

Reply:
only owner of origin (channel) can create assets with remote location, or governance. not local wasm contracts.

Reply:
please keep RFC update with these updates

Reply:
if you do not mind i can specify this?

Reply:
of course, i dont mind.

how will birdge transfers work if user doesnt specify the locaiton if the user creates permissionless asset, during the bridge transfer, what will be the way to match transferred token to the one created by the user. rn match happens by location.